src/search/redis_query_transformer.h

```diff
@@ -141,11 +139,29 @@ struct Transformer : ir::TreeTransformer {
         } else {
           return std::make_unique<ir::AndExpr>(std::move(exprs));
         }
+      } else if (Is<VectorRange>(query)) {
+        auto vector = GET_OR_RET(GetParam(query->children[2]));
+        return std::make_unique<VectorRangeExpr>(std::make_unique<FieldRef>(field),
+                                                 GET_OR_RET(NumberOrParam(query->children[1])),
+                                                 std::make_unique<VectorLiteral>(std::move(vector)));
```

Comment:
here the vector is in binary format, but in SQL it's text format.

We need to unify them.

Reply:
Oh right, this is the todo item left for this PR. I realized I forgot to leave some comments or notes about it—my bad. I’m currently wrapping up the ParseArray functions into parse_util.h. This update should make them more abstract and reusable. I’m aiming to push the new revision soon!